Modularize Foreman API guide (#3353)
theforeman/foreman-documentation: guides/common/modules/proc_creating-linked-lifecycle-environments.adoc

[id="creating-linked-lifecycle-environments"]
= Creating linked lifecycle environments

Use this example to create a path of lifecycle environments.
This procedure uses the default Library environment with ID `1` as the starting point for creating lifecycle environments.

[id="api-creating-linked-lifecycle-environments"]
.API procedure
. Choose an existing lifecycle environment that you want to use as a starting point.

> List the environment by using its ID, in this case, the environment with ID `1`:

Reviewer: ```suggestion
List the environment by using its ID. 
In this case, the environment with ID `1`:
```

----
$ curl \
--header "Accept:application/json" \
--header "Content-Type:application/json" \
--request POST \
--user _My_User_Name_:__My_Password__ \
--data @life-cycle.json \
https://_{foreman-example-com}_/katello/api/environments \
| python3 -m json.tool
----
+
Example response:
+
[source, none, options="nowrap", subs="+quotes,attributes"]
----
      _output omitted_
    "description": null,
    *"id": 2,*
    "label": "api-dev",
    "library": false,
    "name": "API Development",
    "organization": {
        "id": 1,
        "label": "Default_Organization",
        "name": "Default Organization"
    },
    "permissions": {
        "destroy_lifecycle_environments": true,
        "edit_lifecycle_environments": true,
        "promote_or_remove_content_views_to_environments": true,
        "view_lifecycle_environments": true
    },
   *"prior":* {
        *"id": 1,*
        "name": "Library"
    },
    _output truncated_
----
+
In the command output, you can see the ID for this lifecycle environment is `2`, and the lifecycle environment before this one is `1`.
Use the lifecycle environment with ID `2` to create a successor to this environment.

> . Edit the previously created `life-cycle.json` file, updating the `label`, `name`, and `prior` values.

Reviewer: ```suggestion
. Edit the previously created `life-cycle.json` file to update the `label`, `name`, and `prior` values.
```

+
[source, none, options="nowrap", subs="+quotes,attributes"]
----
{"organization_id":1,"label":"api-qa","name":"API QA","prior":2}
----

> . Create a lifecycle environment, by using the `prior` option set to `2`.

Reviewer: ```suggestion
. Create a lifecycle environment using the `prior` option set to `2`.
```
